src/Perception/include/BipedalLocomotion/Perception/Features/ArucoDetector.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file ArucoDetector.h
+ * @authors Prashanth Ramadoss
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PERECEPTION_FEATURES_ARUCO_H
+#define BIPEDAL_LOCOMOTION_PERECEPTION_FEATURES_ARUCO_H
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+#include <Eigen/Dense>
+#include <opencv2/core.hpp>
+
+#include <memory>
+#include <unordered_map>
+
+namespace BipedalLocomotion
+{
+namespace Perception
+{
+
+/**
+ * Aruco marker identifiers
+ */
+struct ArucoMarkerData
+{
+    /**
+     * Marker ID
+     */
+    int id{-1};
+
+    /**
+     * Marker corners in camera coordinates
+     * in the order
+     * - top left
+     * - top right
+     * - bottom right
+     * - bottom left
+     */
+    std::vector<cv::Point2f> corners;
+
+    /**
+     * Pose of the marker in camera frame
+     * cam_H_marker
+     */
+    Eigen::Matrix4d pose;
+};
+
+
+/**
+ * Aruco detector output
+ */
+struct ArucoDetectorOutput
+{
+    std::unordered_map<int, ArucoMarkerData> markers;
+    double timeNow{-1.0};
+};
+
+class ArucoDetector : public System::Advanceable<ArucoDetectorOutput>
+{
+public:
+    ArucoDetector();
+    ~ArucoDetector();
+
+    /**
+     * Initialize the detector
+     * @note The following parameter are required by the filter:
+     * - "marker_dictionary" OpenCV predefined aruco marker dictionary 
+     * (available options: "4X4_50", "4X4_100", "4X4_250", "4X4_1000",
+     *                     "5X5_50", "5X5_100", "5X5_250", "5X5_1000",
+     *                     "6X6_50", "6X6_100", "6X6_250", "6X6_1000",
+     *                     "7X7_50", "7X7_100", "7X7_250", "7X7_1000",
+     *                     "ARUCO_ORIGINAL")
+     * options coherent with https://docs.opencv.org/master/d9/d6a/group__aruco.html#gac84398a9ed9dd01306592dd616c2c975
+     * - "marker_length" marker length in m
+     * - "camera_matrix" 9d vector representing the camera calbration matrix in row major order
+     * - "distortion_coefficients" 5d vector containing camera distortion coefficients
+     * @param[in] handlerWeak weak pointer to a ParametersHandler::IParametersHandler interface
+     * @tparameter Derived particular implementation of the IParameterHandler
+     * @return True in case of success, false otherwise.
+     */
+    bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> handler);
+
+    /**
+     * Set image for which markers need to be detecte
+     * @param[in] inputImg image as OpenCV mat
+     * @param[in] timeNow current time in chosen time units
+     * @return True in case of success, false otherwise
+     */
+    bool setImage(const cv::Mat& inputImg, double timeNow);
```

Comment:
Just by reading the interface and the documentation I cannot understand why the time is needed here.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/159/commits/daa331009be38c957e5c29b015065383907aa6e7